src/ImageBuilder.ts

```diff
@@ -62,13 +62,22 @@ export default class ImageBuilder {
 
             var isCreateBlob = false;
             var imgBuilderId = "";
-            
+
             if (this._taskParameters.customizerSource != undefined && this._taskParameters.customizerSource != "") {
                 isCreateBlob = true;
             }
 
             if (!this._taskParameters.isTemplateJsonProvided) {
-                imgBuilderId = `/subscriptions/${subscriptionId}/resourcegroups/${this._taskParameters.resourceGroupName}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/${this.idenityName}`;
+                if (this.idenityName.indexOf("subscriptions/") > -1) {
+                    imgBuilderId = this.idenityName;
+                }
+                else {
+                    imgBuilderId = `/subscriptions/${subscriptionId}/resourcegroups/${this._taskParameters.resourceGroupName}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/${this.idenityName}`;
+                }
+            }
+            else {
+                var template = JSON.parse(this._taskParameters.templateJsonFromUser);
+                this._taskParameters.location = template.location;
```

Comment:
Why are we doing this? Is location not a mandatory parameter anymore?
What is user gives location as action input and in template too?

Reply:
This will be checked only when user template is not provided and when dist-type is sig

Reply:
Looks incorrect. This is the else block which is for when the user template is provided. We can discuss this on call.

Reply:
ohkay we can discuss it in a call